package/upgrade/upgrade_manifests.sh

```diff
@@ -601,7 +608,18 @@ upgrade_rancher() {
 
   yq -i '.features = "multi-cluster-management=false,multi-cluster-management-agent=false,managed-system-upgrade-controller=false,turtles=false,embedded-cluster-api=true"' values.yaml
 
-  REPO_RANCHER_VERSION=$REPO_RANCHER_VERSION yq -e e '.rancherImageTag = strenv(REPO_RANCHER_VERSION)' values.yaml -i
+  if [[ "$imageMode" == "legacy" ]]; then
```

Comment:
IIUC, we don't need this snippet in the future upgrade paths starting from v1.8.0. Could we add a TODO comment or github issue for the future removal? Not sure how we handle stuff like this before. Thanks.

Reply:
This is a good question.

Indeed, the ugprade code has many legacy functions/patches, they need to be reviewed version by version to clear the out-datede parts.  e.g. https://github.com/harvester/harvester/pull/8939 

meanwhile, we also try best to write the robust code, even when it is not required on new version, it can still run smoothly without any side-effect.